node/src/blockchain/blockchain_interface/data_structures/errors.rs

```diff
@@ -0,0 +1,259 @@
+const BLOCKCHAIN_SERVICE_URL_NOT_SPECIFIED: &str = "Uninitialized blockchain interface. To avoid \
+being delinquency-banned, you should restart the Node with a value for blockchain-service-url";
+
+#[derive(Clone, Debug, PartialEq, Eq, VariantCount)]
+pub enum BlockchainError {
+    InvalidUrl,
+    InvalidAddress,
+    InvalidResponse,
+    QueryFailed(String),
+    UninitializedBlockchainInterface,
+}
+
+impl Display for BlockchainError {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        let err_spec = match self {
+            Self::InvalidUrl => Either::Left("Invalid url"),
+            Self::InvalidAddress => Either::Left("Invalid address"),
+            Self::InvalidResponse => Either::Left("Invalid response"),
+            Self::QueryFailed(msg) => Either::Right(format!("Query failed: {}", msg)),
+            Self::UninitializedBlockchainInterface => {
+                Either::Left(BLOCKCHAIN_SERVICE_URL_NOT_SPECIFIED)
+            }
+        };
+        write!(f, "Blockchain error: {}", err_spec)
+    }
+}
+
+pub type BlockchainResult<T> = Result<T, BlockchainError>;
+pub type ResultForReceipt = BlockchainResult<Option<TransactionReceipt>>;
+
+#[derive(Clone, Debug, PartialEq, Eq, VariantCount)]
+pub enum PayableTransactionError {
+    MissingConsumingWallet,
+    GasPriceQueryFailed(String),
+    TransactionCount(BlockchainError),
+    UnusableWallet(String),
+    Signing(String),
+    Sending { msg: String, hashes: Vec<H256> },
+    UninitializedBlockchainInterface,
+}
+
+impl Display for PayableTransactionError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::MissingConsumingWallet => {
+                write!(f, "Missing consuming wallet to pay payable from")
+            }
+            Self::GasPriceQueryFailed(msg) => {
+                write!(f, "Unsuccessful gas price query: \"{}\"", msg)
+            }
+            Self::TransactionCount(blockchain_err) => {
+                write!(f, "Transaction count fetching failed: {}", blockchain_err)
+            }
+            Self::UnusableWallet(msg) => write!(
+                f,
+                "Unusable wallet for signing payable transactions: \"{}\"",
+                msg
+            ),
+            Self::Signing(msg) => write!(f, "Signing phase: \"{}\"", msg),
+            Self::Sending { msg, hashes } => write!(
+                f,
+                "Sending phase: \"{}\". Signed and hashed transactions: {}",
+                msg,
+                comma_joined_stringifiable(hashes, |hash| format!("{:?}", hash))
+            ),
+            Self::UninitializedBlockchainInterface => {
+                write!(f, "{}", BLOCKCHAIN_SERVICE_URL_NOT_SPECIFIED)
+            }
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, VariantCount)]
+pub enum BlockchainAgentBuildError {
+    GasPrice(PersistentConfigError),
+    TransactionFeeBalance(Wallet, BlockchainError),
+    MasqBalance(Wallet, BlockchainError),
```

Comment:
If we ever get this working on other blockchains, the token there is probably not going to be called MASQ. (Or maybe it is; who knows?) Anyway, there may come a time when we'll need this name to be more general. Is this that time? Maybe not. Okay, never mind--unless you want to.